src/string-util/string-util.ts

```diff
@@ -37,28 +35,70 @@ export namespace StringUtil {
     const start = targetMask[type].start;
     const end = targetMask[type].end;
     const length = type === 'name' ? end - start || 1 : end - start;
-    return getMaskedString({ text, length, maskingStart: start });
+    return getMaskedString({ text, length, maskingStart: start, type });
+
+    function getMaskedString({ text, length, maskingStart, type }: MaskingOpts): string {
+      //emoji가 있는 name에만 붙어서 따로 처리
+      if (type === 'name' && EMOJI_REGEXP.test(text)) {
+        return getMaskedNameAboutEmoji(text);
+      }
 
-    function getMaskedString({ text, length, maskingStart }: MaskingOpts): string {
       const validLength = Math.max(0, length);
       const masking = '*'.repeat(validLength);
       const textSplit = [...text];
       textSplit.splice(maskingStart, validLength, masking);
+
       return textSplit.join('');
     }
 
-    function getMaskingIndexOfUnnormalizedPhone(phone: string): { start: number; end: number } {
-      const phoneNumberParts = phone.split(/\D/);
+    function getMaskedNameAboutEmoji(text: string): string {
+      const splitName = getNameArrayIncludeEmoji(text);
```

Comment:
array의 이름은 복수형태 혹은 ~List/~Array
```suggestion
      const splitNames = getNameArrayIncludeEmoji(text);
```